src/Discord.Net.Commands/CommandService.cs

```diff
@@ -506,19 +506,83 @@ public async Task<IResult> ExecuteAsync(ICommandContext context, string input, I
             services = services ?? EmptyServiceProvider.Instance;
 
             var searchResult = Search(input);
-            if (!searchResult.IsSuccess)
+
+            var validationResult = await ValidateAndGetBestMatch(searchResult, context, services, multiMatchHandling);
+
+            if (validationResult is SearchResult result)
+            {
+                await _commandExecutedEvent.InvokeAsync(Optional.Create<CommandInfo>(), context, result).ConfigureAwait(false);
+                return result;
+            }
+
+            if (validationResult is MatchResult matchResult)
+            {
+                return await handleCommandPipeline(matchResult, context, services);
+            }
+
+            return validationResult;
+        }
+
+        private async Task<IResult> handleCommandPipeline(MatchResult matchResult, ICommandContext context, IServiceProvider services)
```

Comment:
Could you please make this pascal casing?

Reply:
oh sure rider autocorrected it to that.